File: 강병호/5week/[22864] 피로도.py
'''
* Author    : BYEONGHO KANG
* Date      : 2024.11.1(Fri)
* Runtime   : 32 ms
* Memory    : 31120 KB
* Algorithm : 구현
'''

# work per 1h : 피로도 A +, work B -
# 1시간 휴식 : 피로도 C - not minus
# 최대 피로도 M

A, B, C, M = input().split()

A = int(A)
B = int(B)
C = int(C)
M = int(M)

# initialize
fatigue = 0
work = 0

for hour in range(24):
    if fatigue + A <= M:
        fatigue += A
        work += B
    else :
        fatigue = max(0, fatigue - C)
print(work)

File: 이정훈/5주차/[22864] 피로도.cpp
/*
* Author    : Lee JeongHun
* Date      : 2024.11.02(Sat)
* Runtime   : 0 ms
* Memory    : 2020 KB
* Algorithm : Greedy Algorithm
*/

#include <iostream>

using namespace std;

int main()
{
    // 입력 처리
    int a, b, c, m;
    cin >> a >> b >> c >> m;

    // 변수 선언
    int fatigue = 0;
    int work = 0;

    for(int i = 0; i < 24; ++i)
    {
        // 지금 일을 할 수 있는 상태인지 검사
        if(fatigue + a <= m)
        {
            // 일 처리
            fatigue += a;
            work += b;
        }
        else
        {
            // 휴식
            fatigue -= c;
            if(fatigue < 0) fatigue = 0;
        }
    }

    // 정답 출력
    cout << work;

    return 0;
}